File: backend/api/src/main/java/kpn/api/common/PoiState.java
package kpn.api.common;

import kpn.api.custom.Timestamp;
import kpn.core.doc.WithStringId;

import java.util.Optional;

public record PoiState(
  String _id,
  Optional<String> imageLink,
  Optional<String> imageStatus,
  Optional<String> imageStatusDetail,
  Optional<Timestamp> imageFirstSeen,
  Optional<Timestamp> imageLastSeen
) implements WithStringId {
}

File: backend/api/src/main/scala/kpn/api/common/PoiState.scala
package kpn.api.common

import kpn.api.custom.Timestamp
import kpn.api.id.WithStringId

case class PoiState(
  _id: String,
  imageLink: Option[String] = None,
  imageStatus: Option[String] = None,
  imageStatusDetail: Option[String] = None,
  imageFirstSeen: Option[Timestamp] = None,
  imageLastSeen: Option[Timestamp] = None,
) extends WithStringId
